File: Backend/Infrastructure.MyShop/Interfaces/ICommentService.cs
using Data.MyShop.Entities;
using Infrastructure.MyShop.Models.DTO.CommentDTO;
using Infrastructure.MyShop.Services;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Infrastructure.MyShop.Interfaces
{
    public interface ICommentService
    {
        Task<ServiceResponse> GetAllAsync();
        Task<ServiceResponse> CreateCommentAsync(CommentCreateDTO comment);
        Task<ServiceResponse> DeleteCommentAsync(int id);
        Task<ServiceResponse> EditCommentAsync(CommentEditDTO comment);
        Task<ServiceResponse> GetCommentsByProductIdAsync(int id);
    }
}

File: Backend-Java/src/main/java/org/example/Infrastructure/interfaces/IServices/ICommentService.java
package org.example.Infrastructure.interfaces.IServices;

import org.example.Infrastructure.dto.commentDTO.CommentCreateDTO;
import org.example.Infrastructure.dto.commentDTO.CommentEditDTO;
import org.example.Infrastructure.dto.commentDTO.CommentItemDTO;

import java.util.List;

public interface ICommentService {
    CommentItemDTO create(CommentCreateDTO model);
    List<CommentItemDTO> getAll();
    CommentItemDTO edit(CommentEditDTO model);
    void deleteById(int id);
    CommentItemDTO getById(int id);
    List<CommentItemDTO> getCommentsByProductIdAsync(int id);
}
